S7.Net/PLC.cs

```diff
@@ -234,13 +235,34 @@ private static void AssertReadResponse(byte[] s7Data, int dataLength)
 
             if (s7Data.Length < 15) throw NotEnoughBytes();
 
-            if (s7Data[14] != 0xff)
-                throw new PlcException(ErrorCode.ReadData,
-                    $"Invalid response from PLC: '{BitConverter.ToString(s7Data)}'.");
+            ValidateResponseCode((ReadWriteErrorCode)s7Data[14]);
 
             if (s7Data.Length < expectedLength) throw NotEnoughBytes();
         }
 
+        internal static void ValidateResponseCode(ReadWriteErrorCode statusCode)
+        {
+            switch (statusCode)
+            {
+                case ReadWriteErrorCode.ObjectDoesNotExist:
+                    throw new Exception($"Received error from PLC: Object does not exist.");
+                case ReadWriteErrorCode.DataTypeInconsistent:
+                    throw new Exception($"Received error from PLC: Data type inconsistent.");
+                case ReadWriteErrorCode.DataTypeNotSupported:
+                    throw new Exception($"Received error from PLC: Data type not supported.");
+                case ReadWriteErrorCode.AccessingObjectNotAllowed:
+                    throw new Exception($"Received error from PLC: Accessing object not allowed.");
+                case ReadWriteErrorCode.AddressOutOfRange:
+                    throw new Exception($"Received error from PLC: Address out of range.");
+                case ReadWriteErrorCode.HardwareFault:
+                    throw new Exception($"Received error from PLC: Hardware fault.");
```

Comment:
These messages don't need to be interpolated because they don't have any interpolations 😉

Reply:
Resolved in b95b71e